### ISSUE TITLE
Parent ids in checked state

### ISSUE THREAD

**Issue by AliBayatMokhtari** [jakezatecky/react-checkbox-tree#265]
what is the approach or algorithm if we need to store all the selected leaves ids and their parents ids recursively?

**Comment by jakezatecky** [comment]
You can use `checkModel="all"` such that if a parent node has all children fully checked it will be included in the `checked` array.

**Comment by AliBayatMokhtari** [comment]
> You can use `checkModel="all"` such that if a parent node has all children fully checked it will be included in the `checked` array.

Yes but what if we wanna add the parent to the ```checked``` array if at least one of its children is checked?

**Comment by jakezatecky** [comment]
Currently, that is not possible. Partially-checked nodes (that is, nodes with at least one child checked but not all children checked) are never included in the `checked` array.

A future version could support such behavior, but it would require a bit of a rework of the internals.

**Comment by fs86** [comment]
Is there any news on this yet?
